bionic/datatypes.py

```diff
@@ -180,76 +180,63 @@ def __repr__(self):
 
 
 @attr.s(frozen=True)
-class MissingCaseKeyValue:
-    pass
+class MissingCaseKeyToken:
+    def __str__(self):
+        return "<MISSING>"
 
 
 class CaseKey(ImmutableMapping):
     """
-    A collection of name-value pairs that uniquely identifies a case.
+    A collection of name-token pairs that uniquely identifies a case.
     """
 
     # This is a sentinel value used to indicate that no value is available. We can't use
-    # None because None is itself a valid value.
+    # None because None is itself a valid token for None Value.
```

Comment:
Wait, is it true that `None` is a valid token for the `None` value? Doesn't `tokenize(None)` return a string? If so, can we just use `None` as our sentinel value?

Reply:
I really thought `tokenize` returns `None` token for `None` string but you are right, I tried it out and it returns a non-empty token. Turns out `str(None)` returns `"None"` and not `None`. I'll remove `MissingCaseKeyToken`. 

This will change the behavior a little bit since now the token will be `None` instead of `"<MISSING>"`.